### PR TITLE
Paravirtualization support(VirtualBox 5.0-)

### PR DESCRIPTION
**Pull request by densuke**
method paravirtprovider=() added

--paravirtprovider default/legacy/minimal/kvm/hyperv

### CODE REVIEW

**Comment by ovr** [comment]
:+1:

File: plugins/providers/virtualbox/config.rb [hunk 1]
```diff
@@ -107,6 +107,14 @@ def cpus=(count)
...
       end
 
+      # Shortcut for setting para-Virtualization setting for the virtual machine.
+      # Calls #customize internally.
+      #
+      # @param type [String] para-Virtualization type(v5.0: legacy,minimal,hyperv,kvm)
+      def paravirtprovider=(type)
+        customize("pre-boot", ["modifyvm", :id, "--paravirtprovider", type])
+      end
+
```
Comment (luispabon) [inline plugins/providers/virtualbox/config.rb]: Just tested this in conjunction with VBox 5.0. Updated vagrantfile to bring up the box with KVM paravirt, box brought up with KVM just fine :+1:

**Comment by sethvargo** [comment]
Hi @densuke 

Thank you for the Pull Request! Would you mind updating the documentation in the `website/docs` folder to identify this functionality for users?

**Comment by mitchellh** [comment]
We try to reserve top-level provider-specific config to very commonly used configs. Since this isn't very commonly used, please just use the `customize` feature as you are on its own. Thanks!

**Pull request closed without merging**